python_modules/dagster/dagster/core/storage/runs/sql_run_storage.py

```diff
@@ -465,6 +471,23 @@ def add_run_tags(self, run_id: str, new_tags: Dict[str, str]):
                     [dict(run_id=run_id, key=tag, value=new_tags[tag]) for tag in added_tags],
                 )
 
+    def get_latest_run_id_by_step_key(self, step_keys) -> Dict[str, str]:
+        query = (
+            db.select([RunTagsTable.c.run_id, RunTagsTable.c.value])
+            .where(RunTagsTable.c.key == ASSETS_TO_EXECUTE_TAG)
+            .order_by(db.desc(RunTagsTable.c.id))
+        )
+        rows = self.fetchall(query)
```

Comment:
I think this query will fetch every run of an asset job, and then deserialize its asset tags, which is pretty rough perf.